File: algorithms/[283]移动零/solution.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
const moveZeroes = function(nums) {
    const len = nums.length;
    for (let i = len - 1; i >= 0; i -= 1) {
        if (nums[i]) continue;
        nums.push(nums.splice(i, 1));
    }
};

File: algorithms/[283]移动零/solution.py
# topics = ["数组"]

from typing import List


class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        for i in range(len(nums))[::-1]:
            if not nums[i]:
                # pop(i) 的时间复杂度为 0(n)
                nums.append(nums.pop(i))
